nupic/research/monitor_mixin/temporal_memory_monitor_mixin.py

```diff
@@ -327,7 +331,15 @@ def compute(self, activeColumns, sequenceLabel=None, **kwargs):
     # Append last cycle's predictiveCells to *predicTEDCells* trace
     self._mmTraces["predictedCells"].data.append(self.predictiveCells)
 
-    super(TemporalMemoryMonitorMixin, self).compute(activeColumns, **kwargs)
+    # Use the line below for the Python temporal memory (TM)
+    # super(TemporalMemoryMonitorMixin, self).compute(activeColumns, **kwargs)
+
+    # Use the next five lines for the C++ TM
+    array = numpy.array(list(activeColumns), dtype='int32')
+    if kwargs is not None and 'learn' in kwargs:
+      super(TemporalMemoryMonitorMixin, self).compute(len(array), array, kwargs['learn'])
+    else:
+      super(TemporalMemoryMonitorMixin, self).compute(len(array), array, True)
 
```

Comment:
This is ugly! Three differing entry points. I wonder if there is a way to combine them?